packages/compass-preferences-model/src/renderer-ipc.ts

```diff
@@ -0,0 +1,52 @@
+import ipc from 'hadron-ipc';
+import type {
+  GlobalPreferences,
+  UserPreferences,
+  OnPreferenceValueChangedCallback,
+} from './preferences';
+
+/**
+ * API to communicate with preferences from the electron renderer process.
+ */
+export const preferencesIpc = {
+  savePreferences(attributes: Partial<GlobalPreferences>) {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:save-preferences', attributes);
+    }
+    return {} as UserPreferences;
```

Comment:
```suggestion
    return Promise.resolve({});
```

Reply:
But we probably still need to set a type here like:
```ts
return Promise.resolve({} as GlobalPreferences);
```
Otherwise, ts will complain about missing properties of the GlobalPreferences type.

Reply:
Hm yeah … good point. I think I’d go with what you’re suggesting here then :+1: